src/gb/gb.c

```diff
@@ -568,6 +568,13 @@ void GBSkipBIOS(struct GB* gb) {
 		break;
 	}
 
+	// TODO: Emulate randomness of wave RAM in DMG. This is only valid in CGB and AGB.
```

Comment:
Actually, this is wrong, this means that when the emulator starts in DMG mode with the boot ROM enabled the wave RAM will be 0 instead of having the pattern.

Reply:
What is the default pattern on the DMG?

Reply:
https://github.com/LIJI32/SameBoy/blob/72cb391612863a43cb3c876b1d210534a54a4020/Core/gb.c#L1286-L1315

Random, as the TODO comment said.

Reply:
Is that a problem? You're right that it's different from the old behavior though.

Reply:
So the right behaviour would be that ``GBReset()`` always initializes the wave RAM to random values, and ``GBSkipBIOS()`` initializes the GBC/GBA values. However, that changes the behaviour, so I think that in this patch I should just add this loop to ``GBReset()`` unconditionally.

Reply:
Doesn't that just revert what you'd done prior?

Reply:
Well, it's a bit of a revert, but the idea is that I don't want to deal with the DMG randomness of the wave RAM, that's a topic that I don't understand well enough. I guess that the result I'm looking for right now is that the wave RAM has the right values (zero) in the case of GBA in GBA mode, and it behaves the same way as it did before the PR for all other cases.

Reply:
I suppose that's fair for right now. I'd like to merge this before the release since it makes subtle API changes, so you have like two days to finish the PR :P

Reply:
😅

Reply:
Ok, it should be fine now.

Reply:
Honestly at this point the only perceptible difference from the third commit is that...now the memory is always zeroed on AGB (excluding CGB-on-AGB), and there's a more comprehensive comment...but it's in the wrong file now. I'm not sure this helps at all heh.

Reply:
Well, the initial state of the wave RAM has some effect in at least the game R-Type: https://gbdev.gg8.se/wiki/articles/Gameboy_sound_hardware#Power_Control I don't know if there is any GBA game affected by this, though. However, I was testing my code to use the PSG channel 3 and noticed it worked on mGBA but not on hardware, so it's not a stretch to assume that someone else may find the same problem. 

Considering that most people would just assume that it behaves the same way in GBC mode or GBA mode, initializing it to 0 in the GBA is something worth doing. It is a significant difference from the random or square waves of GB mode that, at least, would be heard.

Regarding having it in the wrong files, it depends on how you look at it. The way I see it, it is the responsibility of ``GBSkipBIOS()`` to initialize the pattern in GBC and GBA (GBC mode), not of ``GBAudioReset()`` because it's shared between all GB and GBA modes.

As I said in a previous comment, the right behaviour would be that ``GBReset()`` always initializes the wave RAM to random values (because that's probably what happens, at least on DMG, maybe it changed in GBC but it doesn't make any difference), and ``GBSkipBIOS()`` initializes the GBC/GBA values. However, that would require to randomize the wave RAM in a realistic way.

Reply:
However, if this is going to be that controversial, I can get rid of the third patch of this PR and push it as a different PR.

Reply:
I think that would be a good idea. Plus that one doesn't change API so it can be merged after release if needed.

Reply:
Ok, removed.